src/main/java/org/quiltmc/loader/impl/transformer/RuntimeModRemapper.java

```diff
@@ -61,28 +69,59 @@ final class RuntimeModRemapper {
 	private static final String REMAP_TYPE_STATIC = "static";
 
 	public static void remap(TransformCache cache) {
+		QuiltLauncher launcher = QuiltLauncherBase.getLauncher();
+		MappingConfiguration mappingConfiguration = QuiltLauncherBase.getLauncher().getMappingConfiguration();
+
+		if (mappingConfiguration == null) {
+			// TODO: Right now, there's no easy way for people implementing other game providers
+			// 	 but wanting to use the default Quilt plugin to change the namespaceMappingFrom
+			//   (in fact, they still have to declare a supported-by-Quilt intermediateMappings in their quilt.mod.json,
+			//   even if their game is not obfuscated at all!)
+			//   Once that is supported, this should change to error or even crash when mods we know are not going to
+			//   be able to load are found.
+
+			Log.info(LogCategory.MOD_REMAP, "Not remapping mods because mappings are empty");
+			return;
+		} else if (!mappingConfiguration.getNamespaces().contains("intermediary")) {
+			Log.warn(LogCategory.MOD_REMAP, "Not remapping mods because intermediary is missing!");
```

Comment:
I think you're missing a `return` here, as this will crash if intermediary is actually missing.

```suggestion
			Log.warn(LogCategory.MOD_REMAP, "Not remapping mods because intermediary is missing!");
			return;
```